python/pyspark/sql/connect/streaming/query.py

```diff
@@ -66,11 +69,25 @@ def isActive(self) -> bool:
 
     isActive.__doc__ = PySparkStreamingQuery.isActive.__doc__
 
-    # TODO (SPARK-42960): Implement and uncomment the doc
+    def _execute_await_termination_cmd(self, timeoutMs: Optional[int] = None) -> Optional[bool]:
+        cmd = pb2.StreamingQueryCommand()
+        if timeoutMs is not None:
+            cmd.await_termination.timeout_ms = timeoutMs
+            terminated = self._execute_streaming_query_cmd(cmd).await_termination.terminated
+            return terminated
+        else:
+            cmd.await_termination.no_timeout = True
```

Comment:
We don't really check this field on the server. We don't need to 'no_timeout'. It is implied when `timeout_ms` is not set.